Sources/Parsing/Parser.swift

```diff
@@ -66,9 +66,25 @@ extension Parser {
   where
     Input == Slice<UnsafeBufferPointer<UTF8.CodeUnit>>
   {
-    input.utf8.withContiguousStorageIfAvailable { input -> Output? in
-      var input = input[...]
-      return self.parse(&input)
-    }?.flatMap { $0 }
+    input.utf8
+      .withContiguousStorageIfAvailable { input -> Output? in
+        var input = input[...]
+        return self.parse(&input)
+      }?
+      .flatMap { $0 }
+  }
+
+  @inlinable
+  public func parse<C: Collection>(_ input: C) -> Output?
```

Comment:
This could be pulled out to its own branch, but when benchmarking JSON I wanted to test `Data` as well and this makes it nicer.